sqlglot/dialects/dialect.py

```diff
@@ -331,3 +332,20 @@ def create_with_partitions_sql(self, expression):
             expression.set("this", schema)
 
     return self.create_sql(expression)
+
+
+def generate_tsql_date_delta(self, e):
```

Comment:
Rename this so that we have the _sql suffix like other functions in this file? Also I don't think the "tsql" part is necessary, since this is used by others dialects too and may create confusion later on.

Reply:
I am looking for a way to indicate that this is the 3 parameter version without having to fully type generate_date_delta_three_parameters. Any suggestions?
In my mind the TSQL was to indicate the tsql 3 parameter style, but you are right that it is confusing.

Reply:
How does `date_delta_with_unit_sql` look?

Reply:
I like it, should have thought of that, thank you. as for the placement, I agree this will not be used by many dialects since the other 3 param dialects have different structure as far as I could tell. So only databricks and tsql would use this.